server/etcdserver/api/v3compactor/periodic_test.go

```diff
@@ -244,3 +245,23 @@ func waitOneAction(t *testing.T, r testutil.Recorder) {
 		t.Errorf("expect 1 action, got %v instead", len(actions))
 	}
 }
+
+func waitWithRetry(t *testing.T, compactable *fakeCompactable) ([]testutil.Action, error) {
+	t.Helper()
+
+	var lastErr error
+	var actions []testutil.Action
+
+	expectedActions, maxRetries := 1, 5
+	for retry := 0; retry < maxRetries; retry++ {
+		actions, lastErr = compactable.Wait(expectedActions)
+		if lastErr == nil || len(actions) >= expectedActions {
+			return actions, nil
+		}
+		// Exponential backoff
+		backoffTime := time.Duration(10*(1<<retry)) * time.Millisecond
+		t.Logf("Retry %d: waiting %v before next attempt (last error: %v)", retry+1, backoffTime, lastErr)
```

Comment:
nit: %s is more human readable
```suggestion
		t.Logf("Retry %d: waiting %s before next attempt (last error: %v)", retry+1, backoffTime, lastErr)
```